cocotb/share/lib/embed/gpi_embed.c

```diff
@@ -302,10 +302,20 @@ int embed_sim_init(gpi_sim_info_t *info)
         goto cleanup;
     }
     for (i = 0; i < info->argc; i++) {
-        PyObject *argv_item = PyUnicode_FromString(info->argv[i]);      // New reference
+        PyObject *argv_item_raw = PyBytes_FromString(info->argv[i]);       // New reference
+        if (argv_item_raw == NULL) {
+            PyErr_Print();
+            LOG_ERROR("Unable to create argv item");
```

Comment:
```suggestion
            LOG_ERROR("Unable to create Python bytes object from command line argument %d", i);
```

Reply:
Could do, but there's little value here. The only way this error is reached is if we just printed out `MemoryError` due to having no space left to allocate the object.

Happy to apply that change if you think it's useful though.